lib/Request_Ride_Loc.dart

```diff
@@ -1,40 +1,43 @@
 import 'package:carriage_rider/Request_Ride_Time.dart';
 import 'package:flutter/material.dart';
 import 'package:carriage_rider/Home.dart';
+import 'package:carriage_rider/Ride.dart';
+import 'package:flutter_typeahead/flutter_typeahead.dart';
+import 'package:carriage_rider/AuthProvider.dart';
+import 'package:carriage_rider/LocationsProvider.dart';
+import 'package:carriage_rider/app_config.dart';
+import 'package:provider/provider.dart';
+
 class RequestRideLoc extends StatefulWidget {
+  final Ride ride;
+
+  RequestRideLoc({Key key, this.ride}) : super(key: key);
+
   @override
   _RequestRideLocState createState() => _RequestRideLocState();
 }
 
 class _RequestRideLocState extends State<RequestRideLoc> {
-
-  Widget _buildFromField(){
-    return TextFormField(
-      decoration: InputDecoration(labelText: 'From'),
-    );
-  }
-
-  Widget _buildToField(){
-    return TextFormField(
-      decoration: InputDecoration(labelText: 'To'),
-    );
-  }
+  final _formKey = GlobalKey<FormState>();
+  TextEditingController fromCtrl = TextEditingController();
+  TextEditingController toCtrl = TextEditingController();
 
   final cancelStyle = TextStyle(
-      color: Colors.black,
-      fontWeight: FontWeight.w100,
-      fontSize: 15,
+    color: Colors.black,
+    fontWeight: FontWeight.w100,
+    fontSize: 15,
   );
 
   final questionStyle = TextStyle(
-      color: Colors.black,
-      fontWeight: FontWeight.w800,
-      fontSize: 25,
+    color: Colors.black,
+    fontWeight: FontWeight.w800,
+    fontSize: 25,
   );
```

Comment:
Since these two styles are repeated on every page, it could be beneficial to put them in the TextTheme or a utilities class rather than redefining them on every page; this'll be helpful if the designs get changed (e.g. if the font size changes, we only need to change it in one place and not like 5), and to help make sure we have consistent design (e.g. if one of these was accidentally set to the wrong font size, it wouldn't be consistent)